lib/neurx.ex

```diff
@@ -3,16 +3,126 @@ defmodule Neurx do
   Documentation for Neurx.
   """
 
-  use Application
+  alias Neurx.{Network}
 
-  def start(_type, _args) do
-    import Supervisor.Spec, warn: true
+  @loss_types ["MSE"]
+  @default_loss %{type: "MSE"}
 
-    children = [
-      # Define children/workers and sub-supervisors to be supervised
-      # worker(Neurx.Worker, [arg1, arg2, arg3]), etc..
-    ]
-    opts = [strategy: :one_for_one, name: Neurx.Supervisor]
-    Supervisor.start_link(children, opts)
+  @activation_types ["Sigmoid", "Relu"]
+  @default_activation "Sigmoid"
+
+  @optim_types ["SGD"]
+  @default_optim %{type: "SGD", learning_rate: 0.1}
+  @default_learning_rate 0.1
+
+  @doc """
+  Builds the network.
+  """
+  def build(config) do
+    if config == %{} or config == nil do
```

Comment:
cond needs to always have one match equate to true. If I were to use cond I would need to have a second condition matching True so that no errors are thrown. The if do would be more readable in this case.